src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -205,15 +211,40 @@ private string CreateMergedRuleset(AnalyzerSettings languageSpecificSettings)
                 return languageSpecificSettings.RuleSetFilePath;
             }
 
+            string resolvedRulesetPath = GetAbsoluteRulesetPath();
+
             var mergedRulesetFilePath = Path.Combine(ProjectSpecificOutputDirectory, "merged.ruleset");
             Log.LogMessage(MessageImportance.Low, Resources.AnalyzerSettings_CreatingMergedRuleset, mergedRulesetFilePath);
 
-            var content = GetMergedRuleSetContent(OriginalRulesetFilePath, languageSpecificSettings.RuleSetFilePath);
+            var content = GetMergedRuleSetContent(resolvedRulesetPath, languageSpecificSettings.RuleSetFilePath);
 
             File.WriteAllText(mergedRulesetFilePath, content);
             return mergedRulesetFilePath;
         }
 
+        private string GetAbsoluteRulesetPath()
+        {
+            // If the supplied ruleset path is relative then it is relative to the project folder.
+            // This relative path will be wrong if use it directly in the generated merged ruleset
+            // file so we need to make it absolute.
```

Comment:
I discovered this bug during manual testing. If the user is using one of the default MS-provided rulesets (e.g. _Microsoft Minimum Managed Rules_) then the common targets will pass in an absolute path. If the project is using a custom ruleset then the path can be relative.
I've manually tested with a default ruleset, and a custom ruleset located above and below the project folder, as well as a ruleset that is specified in a separate import.